### PR TITLE
diff: do not emit unified diff for binary files

### PR DESCRIPTION
**Pull request by yuja**
# Checklist

If applicable:
- [x] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [x] I have added tests to cover my changes

### CODE REVIEW

**Review by martinvonz — APPROVED** [review]
> ---/+++ lines are part of unified diff hunks, not Git diff header.

Ah, I suspect that's exactly what I had misunderstood. Thanks for fixing!

**Review by ilyagr — COMMENTED** [review]
Thank you!

~~Re title: do you mean "for executable files"?~~

File: cli/tests/test_diff_command.rs [hunk 1]
```diff
@@ -268,6 +268,146 @@ fn test_diff_empty() {
...
+    new mode 100644
+    index e69de29bb2..0cfbf08886
+    --- a/file1
+    +++ b/file1
+    @@ -1,0 +1,1 @@
+    +2
+    diff --git a/file2 b/file2
+    old mode 100755
+    new mode 100644
+    diff --git a/file3 b/file3
```
Comment (ilyagr) [inline cli/tests/test_diff_command.rs:369]: Do I understand correctly that this is a bug and it is not fixed in this PR?

If so, can you add a comment explaining what the bug is?

**Review by ilyagr — APPROVED** [review]

File: cli/tests/test_diff_command.rs (hunk 1 above)
Comment (yuja) [reply cli/tests/test_diff_command.rs:369]: No, it's not a bug (if the quoted region is displayed correctly.) You can see header-only unified diffs in 5e2a9c274698f4b69b4b58cf226a811edd166e47.

The goal is to disable diff of binary files. I just found a couple of bugs while moving codes around, and I needed some tests covering these.

**Pull request merged**